src/parser.ts

```diff
@@ -0,0 +1,468 @@
+import { alert, output } from "src/store";
+import * as RDFNXParser from "rdf-nx-parser";
+
+function parseQuad(line) {
+  let tokens = RDFNXParser.tokenize(line);
+  if (tokens.length !== 4 && tokens.length !== 5) {
+    console.error(line, tokens);
+    throw new Error(
+      "Expected 3 or 4 tokens for line but found: " + tokens.length
+    );
+  }
+  const lastToken = tokens.pop();
+  if (!lastToken || lastToken.type !== "endOfStatement") {
+    console.error(line, tokens, lastToken);
+    throw new Error("Missing end of statement");
+  }
+  const statement = {
+    subject: tokens[0],
+    predicate: tokens[1],
+    object: tokens[2],
+    graph: tokens[3],
+  };
+  return statement;
+}
+
+interface Pair {
+  type: string;
+  value: string;
+}
+
+function renderLiteral(term): string | Pair {
+  switch (term.datatypeIri) {
+    case "http://www.w3.org/2001/XMLSchema#dateTime":
+      return new Date(term.value).toString();
+    case "http://www.w3.org/1999/02/22-rdf-syntax-ns#JSON":
+      return term.value;
+    case undefined:
+      return term.value;
+  }
+
+  return {
+    type: term.datatypeIri,
+    value: term.value,
+  };
+}
+
+function renderTerm(term): string | Pair {
+  switch (term.type) {
+    case "blankNode":
+      return `_:${term.value}`;
+    case "iri":
+      return term.value;
+    case "literal":
+      return renderLiteral(term);
+    default:
+      return JSON.stringify(term);
+  }
+}
+
+function renderStatement(statement): Array<string | Pair> {
+  const row: Array<string | Pair> = [];
+  row.push(renderTerm(statement.subject));
+  row.push(renderTerm(statement.predicate));
+  row.push(renderTerm(statement.object));
+  if (statement.graph) {
+    row.push(statement.graph);
+  }
+  return row;
+}
+
+export const parse = (vc) => {
+  if (!vc) return;
+  let el;
+  try {
+    el = renderInput(vc.trim());
```

Comment:
Clear the error on successful render: this should synchronize the error state with the content rendering state, addressing the error state ambiguity described in https://github.com/spruceid/tzvm2021-explainer/pull/2#issuecomment-879348114
```suggestion
    el = renderInput(vc.trim());
    alert.set(null);
```
To test:
- Enter a valid signing string input
- Make a change to the input to cause an error
- Use the back button to go back to the valid input: the error should disappear.
- Use the forward button to go to the invalid input again: the error should reappear.
- Use the back button to go back to the valid input again: the error should disappear again.